src/modules/checkIsValidUserInput.js

```diff
@@ -9,7 +9,19 @@
  * @return {boolean} - True if the user input is valid, false otherwise
  */
 function checkIsValidUserInput(userInput) {
-  /* Write your code here */
+  switch (true) {
+    case userInput[0] === '0':
+      return false;
+
+    case isNaN(+userInput):
+      return false;
+
+    case new Set(userInput.split('')).size !== 4:
+      return false;
+
+    default:
+      return true;
+  }
```

Comment:
The switch statement is used here to validate the user input. While it works, using a series of `if` statements might be more readable for this type of validation logic. Consider refactoring for clarity.